L72-core-google_default_credentials-extension.md

```diff
@@ -0,0 +1,58 @@
+Allow Call Credentials to be Specified in `grpc_google_default_credentials_create`
+----
+* Author(s): rbellevi
+* Approver: markdroth
+* Status: Draft
+* Implemented in: Core
+* Last updated: July 6th, 2020
+* Discussion at: https://groups.google.com/g/grpc-io/c/fZNm4pU8e3s/m/2Be8u1n7BQAJ
+
+## Abstract
+
+This document proposes that `grpc_google_default_credentials_create` be
+amended to allow the user to specify their desired call credentials.
+
+## Background
+
+The Google default credentials created by the
+`grpc_google_default_credentials_create` function in Core enable connection to
+Google services via a combination of ALTS and SSL credentials, along with a special oauth2
+token that must assert the same identity as the channel-level ALTS credential.
+
+This oauth2 token attached by `grpc_google_default_credentials_create` is based
+on the default VM service account, but the auth libraries for wrapped languages
+allow the creation of channels with non-default service accounts. In this case,
+a mismatch in identity occurs that may cause an unexpected persistent failure of
+RPCs at runtime.
+
+While the default service account is the correct option is most cases, there
+should be an option that enables auth libraries to supply their own call
+credentials.
+
+## Proposal
+
+I propose that the signature of `grpc_google_default_credentials_create` be
+amended to the following:
+
+```C
+GRPCAPI grpc_channel_credentials*
+grpc_google_default_credentials_create(grpc_call_credentials* call_credentials);
+```
+
+Supplying `nullptr` for `call_credentials` will result in the current behavior
+of the function -- oauth2 call credentials based on the default service account
```

Comment:
we should change wording here -- the current behavior of how google default credentials chooses call credentials is actually based on the "Application Default Credentials" spec (see https://cloud.google.com/docs/authentication/production?_ga=2.68587985.1354052904.1594166352-2074181900.1593114348#finding_credentials_automatically). This  <i>by default</i> authenticates the VM's default service account but isn't necessarily guaranteed to, e.g. if the `GOOGLE_APPLICATION_CREDENTIALS` environment variable is set and points a JSON config file for another identity.